manifest/provider/provider_config.go

```diff
@@ -164,6 +164,28 @@ func GetProviderConfigSchema() *tfprotov5.Schema {
 				DescriptionKind: 0,
 				Deprecated:      false,
 			},
+			{
+				Name:            "ignore_annotations",
```

Comment:
For now `ignore_annotations` is a list of annotations that should be ignored across all created resources, the usage  looks like this:
```hcl
provider "kubernetes" {
  ignore_annotations = [
    "cni.projectcalico.org/podIP",
    "cni.projectcalico.org/podIPs",
    "terraform.io/provider",
  ]
}
```

The provider goes through the list of annotations and does an exact match. Do we want to leave it like this or possibly implement some sort of wildcard mask support? For instance, `keys` -- to cover the above case scenario; `key_prefixes` -- to have the ability to capture all annotations that start with what we have listed here. These are all just possible options. The usage will be something like this(rewrote above example):

```hcl
provider "kubernetes" {
  ignore_annotations {
    keys = [
      "terraform.io/provider",
    ]

    key_prefixes = [
      "cni.projectcalico.org/",
    ]
  }
}
```

Reply:
After the internal discussion we had, I've updated the code to support RegExp expressions as `ignore_annotations` and `ignore_labels` arguments. With RegExp, we cover a lot of potential uses cases, including exact match, prefix and suffix match, and much more.